### PR TITLE
[SE-0143] Enable conditional conformances without a flag

### PR DESCRIPTION
**Pull request by DougGregor**
Remove the `-enable-experimental-conditional-conformances` flag, enabling the feature all the time (and in all language modes). This allows developers to work around any source-compatibility issues due to the rollout of conditional conformances in the standard library (e.g, the collapsed `Slice` types).

To make the limitations of this feature more visible, introduce a runtime warning when a "conforms to protocol" check fails dynamically because we have not yet implemented that behavior for conditional conformances.

### CODE REVIEW

**Comment by DougGregor** [comment]
@swift-ci please test

**Comment by DougGregor** [comment]
@swift-ci please smoke test

File: stdlib/public/runtime/ProtocolConformance.cpp [hunk 1]
```diff
@@ -142,11 +142,46 @@ const {
...
+      break;
+    }
+
+    case TypeMetadataRecordKind::Universal:
+      break;
+    }
+    warning(/*flag=*/0,
+            "warning: Swift runtime does not yet support dynamically "
+            "querying conditional conformance ('%s': '%s')\n",
+            typeName, getProtocol()->Name);
```
Comment (jckarter) [inline stdlib/public/runtime/ProtocolConformance.cpp]: Do you have the original source type metadata pointer from the cast at this point? If so, you can get a presentable string by passing it to `swift_getTypeName`.

Comment (DougGregor) [reply stdlib/public/runtime/ProtocolConformance.cpp]: Ah, I missed `swift_getTypeName()`. I'll try that.

**Comment by DougGregor** [comment]
@swift-ci please smoke test and merge

**Comment by DougGregor** [comment]
@swift-ci please smoke test and merge

**Comment by DougGregor** [comment]
@swift-ci please smoke test and merge

**Comment by DougGregor** [comment]
We get proper demangled names in the warning now.

**Comment by DougGregor** [comment]
@swift-ci please smoke test and merge

**Pull request merged**